arch/risc-v/src/common/riscv_percpu.c

```diff
@@ -86,13 +92,39 @@ void riscv_percpu_init(void)
 
 void riscv_percpu_add_hart(uintptr_t hartid)
 {
-  /* Hart IDs go from 0...4 */
+  riscv_percpu_t *percpu;
+  uintptr_t       stack_top;
+
+  /* Get free entry for this hart, this must not fail */
+
+  percpu = (riscv_percpu_t *)sq_remfirst(&g_freelist);
+  DEBUGASSERT(percpu);
+
+  /* Get interrupt stack (if any) */
+
+  stack_top = 0;
+
+#if CONFIG_ARCH_INTERRUPTSTACK > 15
+#if CONFIG_SMP_NCPUS > 1
+  /* Calculate offset first (stack size * hartid) */
 
-  DEBUGASSERT(hartid < HART_CNT);
+  stack_top = (uintptr_t)&g_intstacktop - (uintptr_t)&g_intstackalloc
+  stack_top = stack_top * hartid;
+  stack_top = (uintptr_t)&g_intstacktop - stack_top;
+#else
+  UNUSED(hartid);
+  stack_top = (uintptr_t)&g_intstacktop;
+#endif
+#endif
+
+  /* Assign hartid and stack top */
+
+  percpu->hartid = hartid;
+  percpu->irq_stack = stack_top;
```

Comment:
If I remove initialization of stack_top from above, won't it put an uninitialized value here ?